stories/utils/generate-widget-stories.js

```diff
@@ -81,27 +82,76 @@ export function generateReportBasedWidgetStories( {
 	group,
 	data,
 	options,
+	additionalVariantCallbacks = {},
 	component: WidgetComponent,
 	wrapWidget = true,
 } ) {
 	const stories = storiesOf( group, module );
 
+	if ( Array.isArray( options ) ) {
+		// 	If options is an array, so must data.
+		if ( ! Array.isArray( data ) ) {
+			throw new Error( 'options is an array, data must be one too' );
+		}
+		// Both must have the same length
+		if ( options.length !== data.length ) {
+			throw new Error( 'options and data must have the same number of items' );
+		}
+	}
+
+	const {
+		Loaded: additionalLoadingCallback,
+		'Data Unavailable': additionalDataUnavailableCallback,
+		Error: additionalErrorCallback,
+	} = additionalVariantCallbacks;
+
 	const variants = {
 		Loaded: ( { dispatch } ) => {
-			dispatch( datastore ).receiveGetReport( data, { options } );
+			if ( Array.isArray( options ) ) {
+				options.forEach( ( option, index ) => {
+					dispatch( datastore ).receiveGetReport( data[ index ], { options: option } );
+				} );
+			} else {
+				dispatch( datastore ).receiveGetReport( data, { options } );
+			}
+
+			// Run any additional callbacks if they exist.
+			if ( additionalVariantCallbacks ) {
+				additionalLoadingCallback( dispatch, data, options );
+			}
 		},
 		'Data Unavailable': ( { dispatch } ) => {
-			dispatch( datastore ).receiveGetReport( [], { options } );
+			if ( Array.isArray( options ) ) {
+				options.forEach( ( option, index ) => {
+					const returnType = Array.isArray( options[ index ] ) ? [] : {};
```

Comment:
This is somewhat off, we need to check the type of `data` here (which is the return value), not `options`, which is always an object and irrelevant for the response type.